File: src/widgets/cpp/status_bar/status_bar.cpp
#include "status_bar.h"

StatusBar::StatusBar(QWidget *parent)
    : QProgressBar(parent), timer(new QTimer(this)), timer_interval(300) {
  connect(timer, &QTimer::timeout, this, [&]() {
    setValue(value() + 1);
    start();
  });
  start();
}

StatusBar::~StatusBar() { delete timer; }

void StatusBar::start() { timer->start(timer_interval); }

void StatusBar::stop() { timer->stop(); }

File: src/widgets/python/status_bar/status_bar.py
from PyQt6 import QtWidgets, QtCore


class StatusBar(QtWidgets.QProgressBar):
    def __init__(self, parent=None):
        super().__init__(parent)
        self.timer = QtCore.QTimer(self)
        self.timer_interval = 300
        self.timer.timeout.connect(self.slot_timer)
        self.start()

    def start(self):
        self.timer.start(self.timer_interval)

    def stop(self):
        self.timer.stop()

    def slot_timer(self):
        self.setValue(self.value() + 1)
